src/input/gpio.rs

```diff
@@ -44,42 +44,36 @@ pub fn decode(ev: &input_event, outer_state: &InputDeviceState) -> Option<InputE
         _ => unreachable!(),
     };
     match ev._type {
-        0 => {
+        ecodes::EV_SYN => {
             /* safely ignored. sync event*/
             None
         }
-        1 => {
-            let (p, before_state) = match ev.code {
-                ecodes::KEY_HOME => (
-                    PhysicalButton::MIDDLE,
-                    state.states[0].fetch_and(ev.value != 0, Ordering::Relaxed),
-                ),
-                ecodes::KEY_LEFT => (
-                    PhysicalButton::LEFT,
-                    state.states[1].fetch_and(ev.value != 0, Ordering::Relaxed),
-                ),
-                ecodes::KEY_RIGHT => (
-                    PhysicalButton::RIGHT,
-                    state.states[2].fetch_and(ev.value != 0, Ordering::Relaxed),
-                ),
-                ecodes::KEY_POWER => (
-                    PhysicalButton::POWER,
-                    state.states[3].fetch_and(ev.value != 0, Ordering::Relaxed),
-                ),
-                ecodes::KEY_WAKEUP => (
-                    PhysicalButton::WAKEUP,
-                    state.states[4].fetch_and(ev.value != 0, Ordering::Relaxed),
-                ),
+        ecodes::EV_KEY => {
+            let p = match ev.code {
+                ecodes::KEY_HOME => {
+                    state.states[0].store(ev.value != 0, Ordering::Relaxed);
+                    PhysicalButton::MIDDLE
+                }
+                ecodes::KEY_LEFT => {
+                    state.states[1].store(ev.value != 0, Ordering::Relaxed);
+                    PhysicalButton::LEFT
+                }
+                ecodes::KEY_RIGHT => {
+                    state.states[2].store(ev.value != 0, Ordering::Relaxed);
+                    PhysicalButton::RIGHT
+                }
+                ecodes::KEY_POWER => {
+                    state.states[3].store(ev.value != 0, Ordering::Relaxed);
+                    PhysicalButton::POWER
+                }
+                ecodes::KEY_WAKEUP => {
+                    state.states[4].store(ev.value != 0, Ordering::Relaxed);
+                    PhysicalButton::WAKEUP
+                }
                 _ => return None,
             };
 
-            // Edge trigger -- debouncing
-            let new_state = ev.value != 0;
-            if new_state == before_state {
-                return None;
-            }
```

Comment:
why deleted?

Reply:
That is part of this issue and explained there: https://github.com/canselcik/libremarkable/pull/47

I didn't separate that, the new input example and this properly which is why I made that issue kinda a given for this one instead of dealing with tons of git merge/rebase conflicts.